persistence/schema/gov.go

```diff
@@ -1,442 +1,145 @@
-		hex.EncodeToString(params.MessageUnpauseServiceNodeFeeOwner),
-		hex.EncodeToString(params.MessageChangeParameterFeeOwner),
-		DefaultBigInt,
-	)
-}
+	subQuery := fmt.Sprintf(`SELECT %s`, fields)
+	//TODO (@deblasis): clarify if and how we should use `enabled` here
+	subQuery += fmt.Sprintf(` FROM %s WHERE name='%s' AND height=%d`, ParamsTableName, paramName, height)
 
-func GetParamQuery(paramName string) string {
-	return fmt.Sprintf(`SELECT %s FROM %s WHERE end_height=%d`, paramName, ParamsTableName, DefaultBigInt)
+	return fmt.Sprintf(`INSERT INTO %s(%s) %s`, ParamsTableName, fields, subQuery)
 }
 
-func GetParamNames() (paramNames []string) {
-	paramNames = make([]string, len(SQLColumnNames))
-	copy(paramNames, SQLColumnNames)
-	return
+func ClearAllGovQuery() string {
+	return fmt.Sprintf(`DELETE FROM %s`, ParamsTableName)
 }
 
-func NullifyParamsQuery(height int64) string {
-	return fmt.Sprintf(`UPDATE %s SET end_height=%d WHERE end_height=%d`, ParamsTableName, height, DefaultBigInt)
+type govParamMetadata struct {
+	PoktValType  string
+	PropertyName string
 }
 
-func SetParam(paramName string, paramValue interface{}, height int64) string {
-	paramNames := GetParamNames()
-	pNamesLen := len(paramNames)
-	var index int
-	// TODO (Team) optimize linear search
-	for i, s := range paramNames {
-		if s == paramName {
-			index = i
+func parseGovProto() (govParamMetadataMap map[string]govParamMetadata) {
+	govParamMetadataMap = make(map[string]govParamMetadata)
+	fields := reflect.VisibleFields(reflect.TypeOf(genesis.Params{}))
+	for _, field := range fields {
+		if !field.IsExported() {
+			continue
 		}
-	}
-	switch v := paramValue.(type) {
-	case int, int32, int64:
-		paramNames[index] = fmt.Sprintf("%d", v)
-	case []byte:
-		paramNames[index] = fmt.Sprintf("'%s'", hex.EncodeToString(v))
-	case string:
-		paramNames[index] = fmt.Sprintf("'%s'", v)
-	default:
-		panic("unknown param value")
-	}
-	subQuery := `SELECT `
-	maxIndex := pNamesLen - 1
-	for i, pn := range paramNames {
-		if i == maxIndex {
-			subQuery += "-1"
-		} else {
-			subQuery += fmt.Sprintf("%s,", pn)
+		poktTag := field.Tag.Get("pokt")
+		protoTag := field.Tag.Get("protobuf")
+		poktValType := extractStructTag(poktTag, "val_type=")
+		protoName := extractStructTag(protoTag, "name=")
+		govParamMetadataMap[protoName] = govParamMetadata{
+			PoktValType:  poktValType,
+			PropertyName: field.Name,
 		}
+		govParamMetadataKeys = append(govParamMetadataKeys, protoName)
 	}
-	subQuery += fmt.Sprintf(` FROM %s WHERE end_height=%d`, ParamsTableName, height)
-	return fmt.Sprintf(`INSERT INTO %s((%s))`, ParamsTableName, subQuery)
+
+	return govParamMetadataMap
 }
 
-func ClearAllGovQuery() string {
-	return fmt.Sprintf(`DELETE FROM %s`, ParamsTableName)
+func extractStructTag(structTag string, key string) string {
+	for len(structTag) > 0 {
+		i := strings.IndexByte(structTag, ',')
+		if i < 0 {
+			i = len(structTag)
+		}
+		switch s := structTag[:i]; {
```

Comment:
Why even have the switch here?

Reply:
Right, I simplified it. It was meant to host additional properties inside the tag but sounds like premature optimization since we might never need more than the current one here